server/data/legacyRestClient.ts

```diff
@@ -0,0 +1,159 @@
+import superagent, { SuperAgentRequest } from 'superagent'
+import Agent, { HttpsAgent } from 'agentkeepalive'
+import { Readable } from 'stream'
+
+import logger from '../log'
+import type { UnsanitisedError } from '../sanitisedError'
+import sanitiseError from '../sanitisedError'
+import { ApiConfig } from '../config'
+
+interface PutPostRequest {
+  path?: string
+  headers?: Record<string, string>
+  responseType?: string
+  data?: unknown
+}
+
+interface GetDeleteRequest {
+  path?: string
+  query?: string
+  headers?: Record<string, string>
+  responseType?: string
+}
+
+interface StreamRequest {
+  path?: string
+  headers?: Record<string, string>
+  errorLogger?: (e: UnsanitisedError) => void
+}
+
+/**
+ * @deprecated This class has been deprecated, and [RestClient] should be used instead for all new code.
+ *
+ * All code currently using this class will be migrated to [RestClient] in due course, and this class will be
+ * removed from the codebase.
+ */
+export default class LegacyRestClient {
```

Comment:
This appears to be a new class, but is actually a rename of the previous `restClient`, but with the additional comment re being deprecated